src/SFML/PCH.hpp

```diff
@@ -62,3 +63,16 @@
 #include <cstdint>
 #include <cstdlib>
 #include <cstring>
+
+
+////////////////////////////////////////////////////////////
+// Precompiled Headers (Test Suite Only)
+////////////////////////////////////////////////////////////
+
+#ifdef SFML_BUILD_TEST_SUITE
+
```

Comment:
Why not just use a separate target_precompiled_headers call within test/CMakeLists.txt? Feels out of place to add non-SFML headers to PCH.hpp.

Reply:
You mean to directly pass `catch_test_macros' instead of `PCH.hpp`?

That was my first approach, but I noticed that Catch2 shares quite a lot of headers in common with SFML (e.g. `ostream`, `algorithm`, `vector`, `memory`), so it made sense to reuse the existing `PCH.hpp` rather than creating a new ad-hoc one for Catch2.

The other issue is that we want `catch_test_macros` to also be part of SFML's own PCH because we use that PCH in our own test suite. 

So keeping everything in the same PCH seemed the most reasonable choice here.

We'd still have to keep `catch2/catch_test_macros` in `SFML`'s own PCH to speed up compilation of our test drivers.

Reply:
> Can’t you have multiple PCH sets? And then reuse the PCH set for the core library alongside the PCH for catch_test_macros?

I don't see how that solves the issue, it would limit PCH reuse unless I am mistaken...

There are basically three categories of targets:

1. SFML library: needs PCH_STDLIB + PCH_SFML
2. SFML tests: needs PCH_STDLIB + PCH_SFML + PCH_CATCH
3. Catch2 library: needs PCH_STDLIB + PCH_CATCH

Right now, my PCH is PCH_STDLIB + PCH_SFML + PCH_CATCH. This allows me to reuse the same PCH for all three categories of targets. 

I definitely want "SFML library" and "SFML tests" to reuse the same PCH, to avoid recompiling two different PCHs -- but that requires that PCH to have access to Catch2 headers anyway. 

If we were to go with your solution, then we'd have a few PCH files:

1. `PCHStdlib.hpp`
2. `PCHStdlibSFML.hpp` (includes `PCHStdlib.hpp`)
3. `PCHStdlibCatch.hpp` (includes `PCHStdlib.hpp`)
4. `PCHStdlibSFMLCatch.hpp` (include `PCHSFML.hpp` and `PCHCatch.hpp`)

- SFML libraries would use `PCHStdlibSFML.hpp`.
- SFML tests would use `PCHStdlibSFMLCatch.hpp`.
- Catch would use `PCHStdlibCatch.hpp`.

I would be able to use `PCHStdlibSFML.hpp` for SFML libraries, and use `REUSE_FROM` from `sfml-system`.

But if I had to use `PCHStdlibSFMLCatch.hpp` for tests I would lose the ability to `REUSE_FROM` from `sfml-system`, which is a step backwards compared to the current approach. The same also applies to SFML examples.

I don't think going down this route would be better... in the end PCHs are an optimization of build times which already heavily screws up header hygiene, I think it's acceptable to use a few `target_include_directories` to make everything work when PCHs are enabled.